### PR TITLE
renaming chore #104

### LINKED ISSUE #104: Rename to delegator and provider
A user (the delegator) delegates to a provider.

- [x] `revoke_msa_delegate` -> `revoke_msa_delegation_by_delegator`
- [x] `delegate_msa_id` -> `provider_msa_id`
- [x] `delegator_msa_id` ✅ 
- [x] Consistent through the codebase and design docs

### PR DESCRIPTION
**Pull request by saraswatpuneet**
A user (the delegator) delegates to a provider.

- [x] `revoke_msa_delegate` -> `revoke_msa_delegation_by_delegator`
- [x] `delegate_msa_id` -> `provider_msa_id`
- [x] Consistent through the codebase and design docs
- [x] Update Delegate to Provider naming convention everyhwere

Tests:
- [x] Run unit tests to ensure everything is passing 

Closes #104

### CODE REVIEW

**Comment by enddynayn** [comment]
Don't forget to update Delegate and Delegator in `DelegateInfoOf`.

File: designdocs/delegation.md [hunk 1]
```diff
@@ -69,12 +69,12 @@ Adds the `MsaId` in the payload as a delegate, to an MSA owning `delegator_msa_i
   * Parameters:
       1. `payload`: authorization data signed by the delegating account
           1. `data` - this is what the MSA owner must sign and provide to the delegate beforehand.
-              * `delegate_msa_id` - the delegate's `MsaId`, i.e. the caller's `MsaId`
```
Comment (saraswatpuneet) [inline designdocs/delegation.md:72]: Generic info:
Delegate is Provider now and accordingly, naming conventions have been updated

**Review by enddynayn — APPROVED** [review]
excellent ⛹️‍♂️

**Comment by enddynayn** [comment]
spec file

**Pull request merged**